scraping/scraper.py

```diff
@@ -127,57 +133,83 @@ def scrape_job_run(workspace, job_run_dict, session, result):
     log.debug(f"Scraping job run in workspace: {workspace.name} "
               f"job ({job_run_dict['job_id']}) "
               f"run id: {job_run_dict['run_id']}")
-    instance = job_run_dict["cluster_instance"]
-    state = job_run_dict["state"]
+    instance = job_run_dict.get("cluster_instance", {})
+    state = job_run_dict.get("state", {})
+    state_life_cycle_state = state.get('life_cycle_state')
+    failed_run = state_life_cycle_state == 'INTERNAL_ERROR'
+
     job_run = JobRun(
         job_id=job_run_dict["job_id"],
         run_id=job_run_dict["run_id"],
         number_in_job=job_run_dict["number_in_job"],
         original_attempt_run_id=job_run_dict["original_attempt_run_id"],
         cluster_spec=job_run_dict["cluster_spec"],
         workspace_id=workspace.id,
-        cluster_instance_id=instance["cluster_id"],
+        cluster_instance_id=instance.get("cluster_id"),
         spark_context_id=instance.get("spark_context_id"),
-        state_life_cycle_state=state["life_cycle_state"],
-        state_result_state=state["result_state"],
+        state_life_cycle_state=state_life_cycle_state,
+        state_result_state=state["result_state"] if not failed_run else 'FAIL',
         state_state_message=state["state_message"],
         task=job_run_dict["task"],
         start_time=to_time(job_run_dict["start_time"]),
         setup_duration=job_run_dict["setup_duration"],
         execution_duration=job_run_dict["execution_duration"],
         cleanup_duration=job_run_dict["cleanup_duration"],
         trigger=job_run_dict["trigger"],
-        creator_user_name=job_run_dict["creator_user_name"],
+        creator_user_name=job_run_dict.get("creator_user_name", "DELETED"),
```

Comment:
This could be the missing part from `job` below as well (the default value).

Reply:
This line should've been erased - probably missed because of some massive undos during modification.